### PR TITLE
Don't save all fetched articles to disk

### PR DESCRIPTION
**Pull request by joewalsh**

### CODE REVIEW

File: Wikipedia/Code/MWKDataStore.m [hunk 1]
```diff
@@ -704,7 +711,7 @@ - (BOOL)migrateToCoreData:(NSError **)error {
 #pragma mark - Memory
 
 - (void)didRecievememoryWarningWithNotifcation:(NSNotification *)note {
```
Comment (montehurd) [inline Wikipedia/Code/MWKDataStore.m]: Oh, would you mind fixing the lower case `m` and 2 spelling errors in this method signature?

Should  be:
`didReceiveMemoryWarningWithNotification`

File: Wikipedia/Code/MWKDataStore.m [hunk 2]
```diff
@@ -1088,60 +1116,63 @@ - (NSArray *)imageInfoForArticleWithURL:(NSURL *)url {
...
+            [allValidArticleKeys addObject:key];
+        }
+        
+        if (allValidArticleKeys.count == 0) {
+            deleteEverythingAndSucceed();
+            return;
+        }
+        
+        dispatch_async(self.cacheRemovalQueue, ^{
+            NSMutableArray<NSURL *> *articleURLsToRemove = [NSMutableArray arrayWithCapacity:10];
```
Comment (montehurd) [inline Wikipedia/Code/MWKDataStore.m]: Curious, why capacity `10`?

File: Wikipedia/Code/WMFDatabaseHouseKeeper.swift [hunk 3]
```diff
@@ -122,6 +122,8 @@ import Foundation
             try moc.save()
         }
         
+        
+
```
Comment (montehurd) [inline Wikipedia/Code/WMFDatabaseHouseKeeper.swift]: Sneaky newlines :)

File: Wikipedia/Code/WMFLegacyDatabaseHouseKeeper.m [hunk 4]
```diff
@@ -1,125 +0,0 @@
-#import "WMFLegacyDatabaseHouseKeeper.h"
```
Comment (montehurd) [inline Wikipedia/Code/WMFLegacyDatabaseHouseKeeper.m]: Woohoo!!!

File: Wikipedia/Code/MWKDataStore.m (hunk 2 above)
Comment (joewalsh) [inline Wikipedia/Code/MWKDataStore.m]: Arbitrary - best guess of average size

**Pull request merged**